### ISSUE TITLE
Alerting based on current alert state (high watermark / low watermark)

### ISSUE THREAD

**Issue by doingitbigdata** [Yelp/elastalert#1004]
Hi,

I'm looking to use metric_aggregation type alerts, but I would like to store the current alert 'state' and have rules based on that state.

An example use case is:
if metric 'transactions' exceeds max_threshold 10,000 for the specified time period, create alert 'transaction overload' and store alert state 'transaction overload'  

then

if alert state = 'transaction overload' and metric 'transactions' is less than min_threshold 8,000, create alert 'transaction overload cleared'

In a nutshell I would like to raise alerts based on a high watermark and then send an alert when the 'bad' state is cleared based on a low watermark. Is this functionality available in elastalert?
